### ISSUE TITLE
Support ubuntu 20.04 LTS

### ISSUE THREAD

**Issue by nickma82** [tuxedocomputers/tuxedo.sh#19]
First: I love the idea of the tuxedo.sh, big fan!

Please add Ubuntu 20.04 (Focal Fossa) support!

(Focal support is missing in the [deb.tuxedocomputers repo](https://deb.tuxedocomputers.com/ubuntu/dists/ as well, standing by until Ubuntu 20.04.01...? )
Best, Nick

**Comment by nickma82** [comment]
Just saw this https://github.com/tuxedocomputers/tuxedo.sh/issues/16#issuecomment-621474890

**Comment by hoergen** [comment]
Oh, that would be great, because it crashed the upgrade from 19.10 to 20.04 on my TUXEDO InfinityBook Pro 15 v4 - SILVER Edition and now it crashed my fresh 20.04 Kubuntu installation, when I activated the additional package repositories. 
Crashed means, it doesn't load the graphical desktop any more, after a new kernel was installed.

Now there are error messages like

`kernel: [    5.077235] i915: `' invalid for parameter `enable_dpcd_backlight'`

```
kernel: [    0.749873] system 00:0a: Plug and Play ACPI device, IDs PNP0c02 (active)
kernel: [    0.750002] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PAGD._STA._OSI], AE_NOT_FOUND (20190816/psargs-330)
kernel: [    0.750016] No Local Variables are initialized for Method [_STA]
kernel: [    0.750017] No Arguments are initialized for method [_STA]
kernel: [    0.750019] ACPI Error: Aborting method \_SB.PAGD._STA due to previous error (AE_NOT_FOUND) (20190816/psparse-529)
kernel: [    0.750336] pnp: PnP ACPI: found 11 devices
kernel: [    0.751876] thermal_sys: Registered thermal governor 'fair_share'
```

```
kernel: [    0.760824] Trying to unpack rootfs image as initramfs...
kernel: [    1.013617] Initramfs unpacking failed: Decoding failed
kernel: [    1.024178] Freeing initrd memory: 80648K
kernel: [    1.024234] DMAR: Intel-IOMMU force enabled due to platform opt in
kernel: [    1.024272] DMAR: No ATSR found
```

```
kernel: [    5.082006] iwlwifi 0000:3a:00.0 wlp58s0: renamed from wlan0
kernel: [    5.118415] snd_hda_intel 0000:00:1f.3: couldn't bind with audio component
kernel: [    5.155732] snd_hda_codec_realtek hdaudioC0D0: autoconfig for ALC293: line_outs=1 (0x14/0x0/0x0/0x0/0x0) type:speaker
kernel: [    5.155736] snd_hda_codec_realtek hdaudioC0D0:    speaker_outs=0 (0x0/0x0/0x0/0x0/0x0)
kernel: [    5.155737] snd_hda_codec_realtek hdaudioC0D0:    hp_outs=1 (0x15/0x0/0x0/0x0/0x0)
kernel: [    5.155738] snd_hda_codec_realtek hdaudioC0D0:    mono: mono_out=0x0
kernel: [    5.155740] snd_hda_codec_realtek hdaudioC0D0:    dig-out=0x1e/0x0
kernel: [    5.155741] snd_hda_codec_realtek hdaudioC0D0:    inputs:
kernel: [    5.155743] snd_hda_codec_realtek hdaudioC0D0:      Front Mic=0x18
kernel: [    5.155744] snd_hda_codec_realtek hdaudioC0D0:      Headset Mic=0x1a
kernel: [    5.155746] snd_hda_codec_realtek hdaudioC0D0:      Internal Mic=0x12
kernel: [    5.221646] snd_hda_codec_hdmi hdaudioC0D2: No i915 binding for Intel HDMI/DP codec
kernel: [    5.221854] hdaudio hdaudioC0D2: Unable to bind the codec
kernel: [    5.222901] input: HDA Intel PCH Front Mic as /devices/pci0000:00/0000:00:1f.3/sound/card0/input14
kernel: [    5.222963] input: HDA Intel PCH Front Headphone as /devices/pci0000:00/0000:00:1f.3/sound/card0/input15
grub-common[744]:  * Recording successful boot for GRUB
apport[728]:  * Starting automatic crash report generation: apport
acpid: starting up with netlink and the input layer
udisksd[777]: udisks daemon version 2.8.4 starting
apport[728]:    ...done.
systemd[1]: Started LSB: automatic crash report generation.
systemd[1]: gpu-manager.service: Succeeded.
systemd[1]: Finished Detect the available GPUs and deal with any system changes.
```

**Comment by nickma82** [comment]
> Crashed means, it doesn't load the graphical desktop any more, after a new kernel was installed.

I also have gdm3 boot problems with all but the initially installed kernel (5.4.0-29-generic). My current workaround is to boot the one explicitly via GRUB Advanced Menu.

**Comment by hoergen** [comment]
Thanks for that hint. In my case only kernel 5.4.0-26-generic is working. But finally it is working YAY

**Comment by nickma82** [comment]
Biggest issue for me is the cpu-governor, which does complete bogus after the first standby (after bootup everything works fine).

* Scales the clock rate down on high CPU load 
```
$cpufreq-info --human|grep "current" 
 current policy: frequency should be within 1.90 GHz and 4.90 GHz.
 current CPU frequency is 0.90 GHz.
```
* always stays at `powersave` no matter what I echo into the scaling governor
`echo "performance" | sudo tee /sys/devices/system/cpu/cpu*/cpufreq/scaling_governor`
Although `cat /sys/devices/system/cpu/cpu0/cpufreq/scaling_available_governors` says `performance powersave`

**Comment by hoergen** [comment]
I needed ti reinstall my computer a second time, because it didn't boot after an update with neither kernel.
After applying tuxedo.sh there are the boot error messages again, but it still boots. Hope the repository will be ready soon for 20.04, because i need to install wine from wineHQ without breaking my computer.

**Comment by vinzv** [comment]
General support for focal/20.04 is there since... a few minutes. 🎉 
Packages are available like [described here](https://www.tuxedocomputers.com/en/Infos/Help-and-Support/Instructions/Add-TUXEDO-Computers-software-package-sources.tuxedo). Update for tuxedo.sh should follow soon.

**Comment by hoergen** [comment]
Thanks so much 👍🌻🎊🕺🚀

**Comment by hoergen** [comment]
Hmm got today a kernel update via apt and it broke my fresh and clean and 2 days old install again. So I ran tuxedo.sh again and it had an "OK" on every like. But my computer doesn't start with kernel 5.4.0-39-generic. But luckily I can start with 5.4.0-37-generic Last time it was a matter of time until i couldn't boot it with any kernel and I had do do a fresh install.

Why do I have to fear that every update can break my standard installation? I was buying a Tuxedo Laptop by purpose to have a fully linux compliant computer, but it is keep making me trouble like no other computer I had before (Thinkpad, Lenovo, Toshiba).

I am totally  annoyed about this situation.

**Comment by vinzv** [comment]
Hey, I get your situation is annoying - but there's one core misunderstanding in place:
Having the repositories ready for 20.04 (as I said above) does not mean `tuxedo.sh` being ready for 20.04. I'm very sorry if I didn't make that clear!

> Hmm got today a kernel update via apt and it broke my fresh and clean and 2 days old install again. So I ran tuxedo.sh again and it had an "OK" on every like. But my computer doesn't start with kernel 5.4.0-39-generic. But luckily I can start with 5.4.0-37-generic Last time it was a matter of time until i couldn't boot it with any kernel and I had do do a fresh install.

Please boot to 5.4.0-37 and uninstall 5.4.0-37 from there.
Then switch to our mirrors in `/etc/apt/sources.list` like this:
```
deb http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal main restricted universe multiverse
deb http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security main restricted universe multiverse
deb http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates main restricted universe multiverse

#deb http://archive.ubuntu.com/ubuntu/ focal main restricted universe multiverse
#deb http://security.ubuntu.com/ubuntu/ focal-security main restricted universe multiverse
#deb http://archive.ubuntu.com/ubuntu/ focal-updates main restricted universe multiverse
```

The intention of these repositories is to keep back updates and give us time for testing. Preinstallations via WebFAI or off the factory already have these repositories activated. Yet in tuxedo.sh the changing is still due to be implemented.

> 
> Why do I have to fear that every update can break my standard installation? I was buying a Tuxedo Laptop by purpose to have a fully linux compliant computer, but it is keep making me trouble like no other computer I had before (Thinkpad, Lenovo, Toshiba).
> 
> I am totally annoyed about this situation.

As said, I fully understand your frustration and we to apologize here. But following the steps above your laptop should be safe and sound!

**Comment by hoergen** [comment]
Thanks so much for your quick reply. I think you meant uninstalling 5.4.0-39 when booted in 5.4.0-37

I removed the 39 er kernel and replaced the mirrors in sources.list

An `apt update` delivers the following error messages

```
Ign:78 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/main amd64 c-n-f Metadata
Ign:89 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/restricted amd64 c-n-f Metadata
Ign:90 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/universe amd64 c-n-f Metadata
Ign:91 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/multiverse amd64 c-n-f Metadata
Ign:92 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/main amd64 c-n-f Metadata
Ign:93 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/restricted amd64 c-n-f Metadata
Ign:94 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/universe amd64 c-n-f Metadata
Ign:95 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 c-n-f Metadata
Ign:20 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/main amd64 c-n-f Metadata
Ign:75 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/restricted amd64 c-n-f Metadata
Ign:76 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/universe amd64 c-n-f Metadata
Ign:77 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/multiverse amd64 c-n-f Metadata
Ign:78 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/main amd64 c-n-f Metadata
Ign:89 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/restricted amd64 c-n-f Metadata
Ign:90 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/universe amd64 c-n-f Metadata
Ign:91 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/multiverse amd64 c-n-f Metadata
Ign:92 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/main amd64 c-n-f Metadata
Ign:93 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/restricted amd64 c-n-f Metadata
Ign:94 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/universe amd64 c-n-f Metadata
Ign:95 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 c-n-f Metadata
Ign:20 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/main amd64 c-n-f Metadata
Ign:75 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/restricted amd64 c-n-f Metadata
Ign:76 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/universe amd64 c-n-f Metadata
Ign:77 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/multiverse amd64 c-n-f Metadata
Ign:78 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/main amd64 c-n-f Metadata
Ign:89 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/restricted amd64 c-n-f Metadata
Ign:90 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/universe amd64 c-n-f Metadata
Ign:91 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/multiverse amd64 c-n-f Metadata
Ign:92 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/main amd64 c-n-f Metadata
Ign:93 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/restricted amd64 c-n-f Metadata
Ign:94 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/universe amd64 c-n-f Metadata
Ign:95 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 c-n-f Metadata
Fehl:20 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/main amd64 c-n-f Metadata
  404  Not Found [IP: 88.99.148.10 80]
Ign:75 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/restricted amd64 c-n-f Metadata
Ign:76 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/universe amd64 c-n-f Metadata
Ign:77 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal/multiverse amd64 c-n-f Metadata
Fehl:78 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/main amd64 c-n-f Metadata
  404  Not Found [IP: 88.99.148.10 80]
Ign:89 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/restricted amd64 c-n-f Metadata
Ign:90 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/universe amd64 c-n-f Metadata
Ign:91 http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security/multiverse amd64 c-n-f Metadata
Fehl:92 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/main amd64 c-n-f Metadata
  404  Not Found [IP: 88.99.148.10 80]
Ign:93 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/restricted amd64 c-n-f Metadata
Ign:94 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/universe amd64 c-n-f Metadata
Ign:95 http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates/multiverse amd64 c-n-f Metadata
Es wurden 586 kB in 3 s geholt (216 kB/s).         
Paketlisten werden gelesen... Fertig
E: Fehlschlag beim Holen von http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu/dists/focal/main/cnf/Commands-amd64 404  Not Found [IP: 88.99.148.10 80]
E: Fehlschlag beim Holen von http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu/dists/focal-security/main/cnf/Commands-amd64 404  Not Found [IP: 88.99.148.10 80]
E: Fehlschlag beim Holen von http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu/dists/focal-updates/main/cnf/Commands-amd64 404  Not Found [IP: 88.99.148.10 80]
E: Einige Indexdateien konnten nicht heruntergeladen werden. Sie wurden ignoriert oder alte an ihrer Stelle benutzt.

```

I deactivated now every other "archive" mirror. I only have 

```
deb http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal main restricted universe multiverse
deb http://mirrors.tuxedocomputers.com/ubuntu/mirror/security.ubuntu.com/ubuntu focal-security main restricted universe multiverse
deb http://mirrors.tuxedocomputers.com/ubuntu/mirror/archive.ubuntu.com/ubuntu focal-updates main restricted universe multiverse
```

I reinstalled 5.4.0-37 with  `apt install --reinstall linux-image-5.4.0-37-generic linux-headers-5.4.0-37 linux-headers-5.4.0-37-generic linux-modules-5.4.0-37-generic linux-modules-extra-5.4.0-37-generic`


And now It doesn't boot into KDE  Desktop any more

Which kernel should i Install?

**Comment by hoergen** [comment]
When I try to remove 5.4.0-37 my system warns me, that there is no other kernel available, And I can't install an older kernel like the 0-26 that is referenced in the tuxedo repostiory.

**Comment by nickma82** [comment]
> And I can't install an older kernel like the 0-26 that is referenced in the tuxedo repostiory.

Why not?
Which kernels are available from the tuxedo debian package repository (`apt-cache search linux-image-5.4`) ?

They'r not done providing full support for 20.04 yet, if you'r not really into it I'd consider waiting a bit until the full 20.04 support is good to go...

**Comment by nickma82** [comment]
A feature review and timeline would be nice though...

**Comment by hoergen** [comment]
I posted some error messages i get, when using the repositories. But anyway I reinstalled the computer again, because by applying everything to tuxedo repositories and managed to install kernel 26 i lost my network connection.

So i hope tuxedo.sh will be available as well to 20.04 and I try to not install any updates until then.

Restore is running ... 700GB

**Comment by hoergen** [comment]
I installed now timeshift snapshot tool and wrote a blog post https://hyperblog.de/hoergen/2020/06/28/timeshift-snapshot-tool (german) about it. Hope this prevents further reinstallations

Edit: And it did very successfully 👍
